mod_reforged/hooks/skills/effects/taunted_effect.nut

```diff
@@ -1,29 +1,73 @@
 ::Reforged.HooksMod.hook("scripts/skills/effects/taunted_effect", function(q) {
+	q.m.DefenseModifier <- 0;
+
 	q.create = @(__original) function()
 	{
 		__original();
-        this.m.Description = "This character is taunted by another character and is much more likely to engage and attack them.";
+		this.m.Description = "This character is taunted by another character and is much more likely to engage and attack them.";
 	}
 
-	q.getTooltip <- function()
+	q.getTooltip = @(__original) function()
 	{
-		local ret = this.skill.getTooltip();
+		local ret = __original();
 
-		if (this.getTauntedTarget() != null)
+		if (this.getTauntSource() != null)
 		{
 			ret.push({
-				id = 10,
+				id = 9,
 				type = "text",
 				icon = "ui/icons/special.png",
-				text = "This character has been taunted by " + ::MSU.Text.colorRed(this.getTauntedTarget().getName())
+				text = "This character has been taunted by " + ::MSU.Text.colorRed(this.getTauntSource().getName())
 			});
+
+			if (this.getMeleeDefenseModifier() != 0)
+			{
+				ret.push({
+					id = 10,
+					type = "text",
+					icon = "ui/icons/melee_defense.png",
+					text = ::Reforged.Mod.Tooltips.parseString(::MSU.Text.colorizeValue(this.getMeleeDefenseModifier()) + " [Melee Defense|Concept.MeleeDefense]")
+				});
+			}
+
+			if (this.getRangedDefenseModifier() != 0)
+			{
+				ret.push({
+					id = 11,
+					type = "text",
+					icon = "ui/icons/ranged_defense.png",
+					text = ::Reforged.Mod.Tooltips.parseString(::MSU.Text.colorizeValue(this.getRangedDefenseModifier()) + " [Ranged Defense|Concept.RangeDefense]")
+				});
+			}
 		}
 
 		return ret;
 	}
 
-	q.getTauntedTarget <- function()	// New helper-function to get the taunter
+	q.onUpdate = @(__original) function( _properties )
+	{
+		__original(_properties);
+
+		if (this.getTauntSource() != null)
+		{
+			_properties.MeleeDefense += this.getMeleeDefenseModifier();
+			_properties.RangedDefense += this.getRangedDefenseModifier();
+		}
+	}
+
+// New Functions
+	q.getTauntSource <- function()
```

Comment:
With the idea of tracking the `taunt source` for this effect, we could "improve" the vanilla effect's behavior to be more sensible in this case:
1. Bro A taunts the target.
2. Bro A dies before target's turn ends.
3. The target still has the taunt effect (it only gets removed on turn end). But in its tooltip it doesn't show anything except its name and description now.
4. Bro B now taunts the target. Now suddenly the same taunted effect shows the new info and also applies its debuffs.

This doesn't look very clean in vanilla, and will look even more unclean in Reforged with these new debuffs which are important to visualize. So I suggest that we follow the following approach:
1. Add a field in `taunted_effect`: `this.m.Source`.
2. Add a field in `taunt.nut` (the active skill): `this.m.Target`.

Then in the `onUse` of taunt, you set `this.m.Target` to the `taunted_effect` in the form of `::MSU.asWeakTableRef`. Similarly in the `taunted_effect` of the target you set `this.m.Source` to the taunt user as weak table ref.

Then in both the `onRemoved` and `onDeath` functions of both skills you remove the `taunted_effect` from the target if the target is alive, and reset the `this.m.Target` of the the taunt skill back to null. Similarly in `onCombatFinished` you set that back to null.

I'll be happy to make this change directly to this branch if you'd rather prefer that I do it.

Reply:
In vanilla one guy can taunt multiple enemies. So the variable should be an array.

This all reminds me on the parent child framework for skills that I suggested a while ago and exemplary implemented in this commit here: https://github.com/Battle-Modders/mod-reforged/commit/1d3d85a6b8ad40c19f38add2af592544864fbc13

I don't want to force such a general system into this specific skill and have to deal with a back and forth of reviews for that.
So if you want it still, please go ahead and do it.

Reply:
I think a simpler solution can be for now to just add an isHidden function which checks for there being a valid taunt source present. That will hide the effect if the taunt source dies before this guy's turn ends - and that achieves the player-side visibility goal. I've added that in a commit.

I've also pushed a "fix" commit that adds check for taunt source being not null via `::MSU.isNull` which is necessary as it's an instance of WeakTableRef, and also that taunt source is alive.